lms/static/scripts/frontend_apps/errors.js

```diff
@@ -20,8 +20,7 @@ export class APIError extends Error {
    *   @param {any} [data.details]
    */
   constructor(status, data) {
-    // If message is omitted, pass a default error message.
-    const message = data.message || 'API call failed';
+    const message = data.message || '';
```

Comment:
This is the crux of these changes.

Reply:
With the proposed change, the `APIError` class is being changed to resolve a presentation concern in `ErrorDisplay`. The other way to deal with this might be to give `ErrorDisplay` special handling of the passed error is an `APIError` instance, where it would use the `errorMessage` property rather than `message`. It depends on what is appropriate handling for the `message` property for other kinds of error in `ErrorDisplay`.

Reply:
Yes, true, and I suppose what you're suggesting is an option. I just dislike having to make a generic UI component have to have switching logic based on the type of error it gets. It bugs me.